### ISSUE TITLE
Incorrect AUC value in CatBoost chart with sample_weight

### ISSUE THREAD

**Issue by off99555** [mljar/mljar-supervised#538]
I trained a dataset with sample weight using 3 algos: LightGBM, Xgboost, and CatBoost.
I found that the learning curve chart for CatBoost doesn't take into account the sample weight but the score in the table does.
Maybe you forgot to put sample_weight for CatBoost charts? I also see the problems in the ROC curve chart.
Also, could this affect the training result e.g. terminating at the wrong place? Because I saw the model trained for too many iterations.

![image](https://user-images.githubusercontent.com/15215732/167322423-1b70896b-890f-493e-ae7c-55bd1c7790c7.png)
![image](https://user-images.githubusercontent.com/15215732/167322603-542f692e-c088-4a4a-bd7b-f36720fb9d38.png)

**Comment by pplonski** [comment]
@off99555 thank you for reporting. Is it the problem only for CatBoost?

**Comment by off99555** [comment]
@pplonski Yes, I think so. At least LightGBM and Xgboost don't seem to have this problem.

**Comment by alencn1024** [comment]
kindly to ask is there somebody working on this issue? If not, I'm glad to undertake it @pplonski

**Comment by pplonski** [comment]
@alencn1024 thanks for looking into it!

**Comment by neelima-bhaktha** [comment]
@pplonski sir, 
is anybody working on this issue? if no, i would like to take it up.
